### PR TITLE
Fix HuBERT positional batch norm padding handling

### LINKED ISSUE #47739: HuBERT positional convolution is sensitive to padding when conv_pos_batch_norm=True
I noticed a potential padding-handling issue in HuBERT when `conv_pos_batch_norm=True`.

With this configuration, `HubertPositionalConvEmbedding` applies `BatchNorm1d` without accounting for the padded time steps. 

See [modeling_hubert: L85](https://github.com/huggingface/transformers/blob/main/src/transformers/models/hubert/modeling_hubert.py#L85)

This has two consequences:

1. Padding positions no longer remain zero after batch normalization.
2. [only in training mode] The normalized representations of valid time steps are influenced according to the padding.

As a result, the output for a given sample may change depending on the lengths of the other samples in the same batch, even when the sample itself is unchanged.

I believe this behavior might be in other models as well. So it's worth checking for a similar pattern.

A quick fix would be to at least zero out the padded time frames again after normalization by passing the attention mask, but the second consequence still remains.

Here's a quick example with output that shows the difference:

```python
import torch
from transformers import HubertConfig
from transformers.models.hubert.modeling_hubert import (
    HubertPositionalConvEmbedding,
)
from torch.nn.utils.rnn import pad_sequence

config = HubertConfig(
    hidden_size=1,
    num_conv_pos_embeddings=3,
    num_conv_pos_embedding_groups=1,
    feat_extract_activation="linear",
    conv_pos_batch_norm=True,
)

layer = HubertPositionalConvEmbedding(config).eval()

with torch.no_grad():
    layer.batch_norm.bias.normal_() # bias is initialized to 0, so we need to set it to something else to see the effect of batch norm

sz = torch.randint(1, 10, (1,)).item()

sample1 = torch.randn(sz, 1)
sample2 = torch.randn(2 * sz, 1) # Longer sequence to pad the first sample to.

batch = pad_sequence([sample1, sample2], batch_first=True)

output_alone = layer(sample1.unsqueeze(0))
output_batched = layer(batch)[0, :len(sample1)]

print(output_alone)
print(output_batched)
print((output_alone - output_batched).abs().mean().item())  # Non-negligible difference.
```

### PR DESCRIPTION
**Pull request by kunwar-vikrant**
[CI](https://transformers-ci.lor-e.huggingface.cool/d/pytest-observability-by-pr/pytest-observability-branch?var-pr=47765)

# What does this PR do?
When `conv_pos_batch_norm=True`, HuBERT includes padded time steps in BatchNorm statistics. BatchNorm may also turn zero-padded positions into non-zero values, which then affect valid boundary tokens through the positional convolution.

- Passes the original 2D attention mask to `HubertPositionalConvEmbedding`.
- Excludes padded positions from BatchNorm statistics during training.
- Zeros padded positions before the positional convolution.
- Supports both standard and stable-layer-norm HuBERT encoders.
- Makes the changes in `modular_hubert.py` and regenerates `modeling_hubert.py`.
- Adds regression coverage for both training and evaluation modes.

### Tests

- HuBERT test suite: 188 passed, 195 skipped.
- Modular conversion consistency check passed.
- Code style checks passed.

Fixes #47739

### CODE REVIEW

**Comment by Abdulrhman-shaheen** [comment]
please, read my comment on https://github.com/huggingface/transformers/pull/47763

**Comment by kunwar-vikrant** [comment]
@Abdulrhman-shaheen here modeling_hubert.py is generated using modular_hubert.py which is the source of truth.

**Comment by Rocketknight1** [comment]
No code agent PRs from first time contributors please!
